homeassistant/components/isy994/entity.py

```diff
@@ -69,76 +73,36 @@ def async_on_control(self, event: NodeProperty) -> None:
-            if node.parent_node is not None:
-                # This is not the parent node, get the parent node.
-                node = node.parent_node
-                basename = node.name
-        else:
-            # Default to the hub device if parent node is not a physical device
-            return DeviceInfo(identifiers={(DOMAIN, uuid)})
-
-        device_info = DeviceInfo(
-            identifiers={(DOMAIN, f"{uuid}_{node.address}")},
-            manufacturer=node.protocol,
-            name=f"{basename} ({(str(node.address).rpartition(' ')[0] or node.address)})",
-            via_device=(DOMAIN, uuid),
-            configuration_url=url,
-            suggested_area=node.folder,
-        )
-
-        # ISYv5 Device Types can provide model and manufacturer
-        model: str = "Unknown"
-        if node.node_def_id is not None:
-            model = str(node.node_def_id)
-
-        # Numerical Device Type
-        if node.type is not None:
-            model += f" ({node.type})"
-
-        # Get extra information for Z-Wave Devices
-        if node.protocol == PROTO_ZWAVE:
-            device_info[ATTR_MANUFACTURER] = f"Z-Wave MfrID:{node.zwave_props.mfr_id}"
-            model += (
-                f" Type:{node.zwave_props.devtype_gen} "
-                f"ProductTypeID:{node.zwave_props.prod_type_id} "
-                f"ProductID:{node.zwave_props.product_id}"
-            )
-        device_info[ATTR_MODEL] = model
-
-        return device_info
-
-    @property
-    def unique_id(self) -> str | None:
-        """Get the unique identifier of the device."""
-        if hasattr(self._node, "address"):
-            return f"{self._node.isy.configuration[ISY_CONF_UUID]}_{self._node.address}"
-        return None
-
-    @property
-    def name(self) -> str:
-        """Get the name of the device."""
-        return self._name or str(self._node.name)
-
 
 class ISYNodeEntity(ISYEntity):
     """Representation of a ISY Nodebase (Node/Group) entity."""
 
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Return the device info."""
+        assert self.platform and self.platform.config_entry
+
+        entry_id = self.platform.config_entry.entry_id
+        device_info: dict[str, DeviceInfo] = self.hass.data[DOMAIN][entry_id][
+            ISY_DEVICES
+        ]
+        primary_address = self._node.primary_node
+
+        if self._node.protocol == PROTO_GROUP:
+            if len(self._node.controllers) == 1:
+                # If Group has only 1 Controller, link to that device instead of the hub
+                primary_address = self._node.isy.nodes.get_by_id(
+                    self._node.controllers[0]
+                ).primary_node
+            else:
+                return self._attr_device_info
+
+        if device := device_info.get(primary_address):
+            return device
+
+        # Fall back to the hub if we don't have a primary device
+        return self._attr_device_info
```

Comment:
It would be a bit cleaner to set `self._attr_device_info` in `__init__`

Reply:
Backed it all the way up to `ISYEntity __init__` pass from each platforms' `async_setup_entry`. I had been avoided it since `binary_sensor` is a mess.